### PR TITLE
Making !param a modal link

### LINKED ISSUE #17236: Make parameters in MooseDocs linkable
## Reason
Currently `[!param](a dummy parameter)` in MooseDocs makes the parameter included in the command display with its name and a floating description for the parameter when the mouse is on it. It is sometimes desirable to see the full doc of the parameter including its type, default values, notes, etc.

## Design
So I suggest to let `!param` also add a link to the parameter name, so that users can click the link for more information about the parameter.

## Impact
Make users easier to find detailed parameter documentations.

### PR DESCRIPTION
**Pull request by zachmprince**
This PR changes the behavior of the `!param` command for MooseDocs. It is now a link that opens a pop up window with the detailed parameter information that is seen wen doing `!syntax parameters`. Here is an example for `[!param](/Executioner/Steady/solve_type)`:

image

Closes #17236

### CODE REVIEW

**Comment by loganharbour** [comment]
Can we really not separate that stuff with commas? At least the options for MooseEnum?

**Comment by YaqiWang** [comment]
For MooseEnum, suggest to use `|` or `/`.

**Comment by zachmprince** [comment]
So this is exactly what the syntax is when doing `!syntax parameters`. So I would said it isn't really relevant to this PR, but if we can agree on something, I can implement it really quickly. We hack the command a bit in Griffin and this is what that looks like:

image

**Comment by YaqiWang** [comment]
Sounds good to me. I agree it is irrelevant to this PR.

**Comment by aeslaughter** [comment]
This works for me, but you need to add tests.
https://github.com/idaholab/moose/blob/next/python/MooseDocs/test/extensions/test_appsyntax.py

File: python/MooseDocs/extensions/appsyntax.py [hunk 1]
```diff
@@ -689,87 +691,108 @@ def _createHTMLHelper(self, parent, token, page, func, collapse):
...
+
+        cpp_type = param['cpp_type']
+        p = html.Tag(body, 'p', class_='moose-parameter-description-cpptype')
+        html.Tag(p, 'span', string='C++ Type:')
+        html.String(p, content=cpp_type, escape=True)
+
+        if 'options' in param:
+            p = html.Tag(body, 'p', class_='moose-parameter-description-options')
+            html.Tag(p, 'span', string='Options:')
+            html.String(p, content=param['options'])
```
Comment (aeslaughter) [inline python/MooseDocs/extensions/appsyntax.py:767]: I vote for a comma between the options.
```suggestion
            html.String(p, content=", ".join(param['options'].split()))
```

**Review by aeslaughter — DISMISSED** [review]
Looks good to me, I am going to start pushing MooseDocs tickets to you now. 

If you want to add the comma thing go for it, I also don't mind the spaces. We don't use commas in the input file.

File: python/MooseDocs/extensions/appsyntax.py (hunk 1 above)
Comment (zachmprince) [reply python/MooseDocs/extensions/appsyntax.py]: Thanks Andrew. I went ahead and added the commas.

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**